js/migration/components/steps/migrate-blocks.js

```diff
@@ -70,16 +70,22 @@ const MigrateBlocks = ( { isStepActive, isStepComplete, stepIndex } ) => {
 	 * Migrates the post content.
 	 */
 	const migratePostContent = async () => {
+		// Used for a 504 Gateway Timeout Error, but could also be for other errors.
+		const timeoutErrorCode = 'invalid_json';
+
 		await apiFetch( {
 			path: '/block-lab/migrate-post-content',
 			method: 'POST',
 		} ).then( () => {
 			speak( __( 'The migration was successful!', 'block-lab' ) );
 			setIsSuccess( true );
-		} ).catch( ( result ) => {
-			if ( result.hasOwnProperty( 'message' ) ) {
+		} ).catch( async ( result ) => {
+			if ( result.hasOwnProperty( 'code' ) && timeoutErrorCode === result.code ) {
+				await migratePostContent();
```

Comment:
Migrating ~10,000 posts took about 8 minutes, and the REST API request to migrate the `post_content` timed out after a minute:

![timed-out](https://user-images.githubusercontent.com/4063887/88616378-a3faa280-d059-11ea-9ac1-56ddbbf05410.png)

So this looks for the `invalid_json` code that a timeout has, and makes the request again. 

The user won't see that the request failed and that it's making a new one.

There's some risk that something else causes an `invalid_json` code, and this keeps looping and never shows an error message.

But the tradeoff is that we can handle migrating thousands of posts.

In my local, it migrated about 1250 posts per minute.